rust/lance-linalg/src/distance/cosine.rs

```diff
@@ -227,7 +227,7 @@ where
     let xy = dot(x, y);
     // 1 - xy / (sqrt(x_sq) * sqrt(y_sq))
     // use f64 for overflow protection.
-    1.0 - (xy / (x_norm * y_norm + f32::epsilon()))
+    1.0 - (xy / x_norm * y_norm)
```

Comment:
```suggestion
    1.0 - (xy / (x_norm * y_norm))
```

Reply:
Good catch!

Reply:
Make this a reminder that you should be adding many more tests to validate your math is correct :)